pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -1892,3 +1892,213 @@ func createChainedEcOpHinter(resolver hintReferenceResolver) (hinter.Hinter, err
 
 	return newChainedEcOpHint(len, p, m, q, s), nil
 }
+
+// BigIntPackDivMod hint divides two values modulo a prime number
+//
+// `newBigIntPackDivModHint` takes 3 operanders as arguments
+//   - `P` is the prime modulus
+//   - `x` is the numerator
+//   - `y` is the denominator
+//
+// `newBigIntPackDivModHint` assigns the result as `value` in the current scope
+func newBigIntPackDivModHint(x, y, p hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "DivMod",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.cairo_secp.secp_utils import pack
+			//> from starkware.cairo.common.math_utils import as_int
+			//> from starkware.python.math_utils import div_mod, safe_div
+			//>
+			//> p = pack(ids.P, PRIME)
+			//> x = pack(ids.x, PRIME) + as_int(ids.x.d3, PRIME) * ids.BASE ** 3 + as_int(ids.x.d4, PRIME) * ids.BASE ** 4
+			//> y = pack(ids.y, PRIME)
+			//>
+			//> value = res = div_mod(x, y, p)
+
+			pAddr, err := p.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			pValues, err := vm.Memory.ResolveAsBigInt3(pAddr)
+			if err != nil {
+				return err
+			}
+
+			xAddr, err := x.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			xValues, err := vm.Memory.ResolveAsBigInt3(xAddr)
+			if err != nil {
+				return err
+			}
+
+			yAddr, err := y.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			yValues, err := vm.Memory.ResolveAsBigInt3(yAddr)
+			if err != nil {
+				return err
+			}
+
+			var xD3Big big.Int
+			var xD4Big big.Int
+
+			xValues[0].BigInt(&xD3Big)
```

Comment:
why do you assign xValues[0] and xValues[1] ?

dont you need xValues[3] and xValues[4] ?

`x` is a `UnreducedBigInt5` which is : 

```
struct UnreducedBigInt5 {
    d0: felt,
    d1: felt,
    d2: felt,
    d3: felt,
    d4: felt,
}
```

resolving`x` as a `BigInt3` wont work here because we need to access 4th and 5th fields here

you should create e new `ResolveAsBigInt3` function to retrieve the values you need